registration/include/pcl/registration/meta_icp.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2015, Michael 'v4hn' Goerner
+ *  Copyright (c) 2015-, Open Perception, Inc.
+ *
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder(s) nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef PCL_REGISTRATION_META_ICP_H_
+#define PCL_REGISTRATION_META_ICP_H_
+
+#include <pcl/point_cloud.h>
+#include <pcl/registration/registration.h>
+
+namespace pcl {
+  namespace registration {
+
+    /** \brief Meta @ref IterativeClosestPoint class
+      *
+      * This class provides a way to register a stream of clouds where each cloud
+      * will be aligned to the conglomerate of all previous clouds.
+      *
+      * \code
+      * IterativeClosestPoint<PointXYZ,PointXYZ>::Ptr icp (new IterativeClosestPoint<PointXYZ,PointXYZ>);
+      * icp->setMaxCorrespondenceDistance (0.05);
+      * icp->setMaximumIterations (50);
+      *
+      * MetaICP<PointXYZ> micp;
+      * micp.setICP (icp);
+      *
+      * while (true){
+      *   PointCloud<PointXYZ>::Ptr cloud (new PointCloud<PointXYZ>);
+      *   read_cloud (*cloud);
+      *   micp.registerCloud (cloud);
+      *
+      *   PointCloud<PointXYZ>::Ptr tmp (new PointCloud<PointXYZ>);
+      *   transformPointCloud (*cloud, *tmp, micp.getAbsoluteTransform ());
+      *   write_cloud (*tmp);
+      * }
+      * \endcode
+      *
```

Comment:
Maybe add a note that a meta cloud is being accumulated? I think it's important for users to be aware of so that there is no surprise where all the memory went.